### ISSUE TITLE
Following OWL 2 Web Ontology Language Structural Specification and Functional-Style Syntax Documentation

### ISSUE THREAD

**Issue by Demirrr** [dice-group/owlapy#89]
The interface of owlapy should strictily follow [OWL 2 Web Ontology Language Structural Specification and Functional-Style Syntax ](https://www.w3.org/TR/owl2-syntax/).
This would help increase the user satisfaction, e.g., less time in learning how the library works.

For instance, I guess owlapy.class_expression should follow [the structure of Class Expressions section](https://www.w3.org/TR/owl2-syntax/#Class_Expressions).
The Class Expression definition provided in the recommendation
```
ClassExpression :=
    Class |
    ObjectIntersectionOf | ObjectUnionOf | ObjectComplementOf | ObjectOneOf |
    ObjectSomeValuesFrom | ObjectAllValuesFrom | ObjectHasValue | ObjectHasSelf |
    ObjectMinCardinality | ObjectMaxCardinality | ObjectExactCardinality |
    DataSomeValuesFrom | DataAllValuesFrom | DataHasValue |
    DataMinCardinality | DataMaxCardinality | DataExactCardinality
```
would lead to the following
```python
from owlapy.class_expression import OWLClass, ObjectIntersectionOf, ObjectUnionOf, ObjectComplementOf, ObjectOneOf, ObjectSomeValuesFrom, ObjectAllValuesFrom, ObjectHasValue, ObjectHasSelf, ObjectMinCardinality, ObjectMaxCardinality, ObjectExactCardinality, DataSomeValuesFrom, DataAllValuesFrom, DataHasValue, DataMinCardinality, DataMaxCardinality, DataExactCardinality
```

Is there any downside of this ? @alkidbaci @LckyLke ?

**Comment by alkidbaci** [comment]
As much as we have tried to adhere to the architecture in OWL 2 Specification, it becomes a problem if we strictly follow it. If we were to have the exact inheritance hierarchy that is used there it will certainly lead to circular import issues. The diagrams there intersect with each other making it even more difficult to represent the exact structure in code and at the same time have them separated in different modules (which again, will lead to circular import issues). I think the structure can be slightly changed to cohere with OWL 2 Specification architecture a little bit more but not exactly the same.

**Comment by Demirrr** [comment]
Alrighty. I will take care of it

**Comment by Demirrr** [comment]
I think we have already done it. Shall we close the issue ?

**Comment by alkidbaci** [comment]
Yes I believe there is not much changes to do in this direction and if any, then we can track it specifically on a new issue.
